### PR TITLE
refactor garph.ErrData

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

mostly quality of life, but also some error handling
functionality.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #1970

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by dadams39 — APPROVED** [review]
A few nits, but nothing that should get in the way

File: src/internal/connector/graph/errors.go [hunk 1]
```diff
@@ -40,6 +43,17 @@ var (
...
 
+var (
+	mysiteURLNotFound = "unable to retrieve user's mysite URL"
+	mysiteNotFound    = "user's mysite not found"
+)
+
+var Labels = struct {
+	MysiteNotFound string
+}{
+	MysiteNotFound: "mysite_not_found",
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/graph/errors.go:56]: The intent is for this to act as the const.  Difference (in usage) between `graph.LabelMysiteNotFound` and `graph.Labels.MysiteNotFound`.  I'm open to not keeping with this pattern, knowing that it isn't always a favorite.

File: src/internal/connector/graph/errors.go [hunk 2]
```diff
@@ -179,43 +193,74 @@ func hasErrorCode(err error, codes ...string) bool {
...
+}
+
+func setLabels(err *clues.Err, msg string) *clues.Err {
+	if strings.Contains(msg, mysiteNotFound) || strings.Contains(msg, mysiteURLNotFound) {
+		err = err.Label(Labels.MysiteNotFound)
+	}
+
+	return err
+}
+
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/graph/errors.go:248]: Ah, that shouldn't be exported.

File: src/internal/connector/graph/errors.go [hunk 3]
```diff
@@ -179,43 +193,74 @@ func hasErrorCode(err error, codes ...string) bool {
...
+	return setLabels(clues.Stack(e).WithClues(ctx).With(data...), innerMsg)
+}
+
+func setLabels(err *clues.Err, msg string) *clues.Err {
+	if strings.Contains(msg, mysiteNotFound) || strings.Contains(msg, mysiteURLNotFound) {
+		err = err.Label(Labels.MysiteNotFound)
+	}
+
+	return err
+}
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/graph/errors.go:247]: well, yes, it's both.

**Pull request merged**